src/components/ProgressBar/ProgressBar.jsx

```diff
@@ -1,29 +1,77 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import dispatchAnalyticsEvent from '../../helpers/analytics';
 
-export default function ProgressBar({ percent, label }) {
-  return (
-    <div
-      className="progress-bar"
-      role="progressbar"
-      aria-valuenow={percent}
-      aria-valuemin="0"
-      aria-valuemax="100"
-      tabIndex="0"
-      aria-label={label}
-    >
-      <div className="progress-bar-inner" style={{ width: `${percent}%` }} />
-    </div>
-  );
+class ProgressBar extends Component {
```

Comment:
Why was this turned into a class-based component? Did an effect hook not work?

Reply:
Nope - I didn't see any examples of `useEffect` in the component-library, so just decided to switch to a class-based component to take advantage of the lifecycle. 

Want me to convert this over to a hook?

Reply:
Yeah, if it can be done with a hook that would be ideal :+1:

Reply:
Sure thing.  Look OK to you?

https://github.com/department-of-veterans-affairs/component-library/blob/bba37848873b57e8108bc886ff7934085836661a/src/components/ProgressBar/ProgressBar.jsx